src/devices/machine/mb8421.h

```diff
@@ -94,27 +94,34 @@ class mb8421_master_device : public device_t
 	// internal helpers
 	template<read_or_write row, bool is_right> void update_intr(offs_t offset);
 
+  u64 m_data_mask; // for 9/18/36bit rams
+  size_t m_ram_size;
+  size_t m_ram_mask;
+  size_t m_int_addr_left;
+  size_t m_int_addr_right;
+
 private:
 	devcb_write_line m_intl_callback;
 	devcb_write_line m_intr_callback;
 };
 
-// ======================> mb8421_device
+// ======================> mb8421_master_8bit_device
 
-class mb8421_device : public mb8421_master_device
+class mb8421_master_8bit_device : public mb8421_master_device
```

Comment:
Abstract things that derive from `device_t` have names ending in `device_base` by convention.